### PR TITLE
Catalog annotation

### PR DESCRIPTION
**Pull request by jrchudy**
This PR looks for annotations defined on the catalog object in the json schema document. It should be merged alongside the corresponding [PR in chaise](https://github.com/informatics-isi-edu/chaise/pull/1758).

This PR has changes to resolve [#1704 in chaise](https://github.com/informatics-isi-edu/chaise/issues/1704).

### CODE REVIEW

File: docs/user-docs/annotation.md [hunk 1]
```diff
@@ -748,6 +749,23 @@ If a column-level annotation sets a _preference_ of `null`, this suppresses any
 
 This annotation is a hint to ERMrest during table or column creation, when indexes are built. Therefore, administrators SHOULD supply the annotation within table or column creation requests. Manipulation of the annotation on existing tables or columns will not change the indexes which are already present (or absent) on those existing models. However, changes to the table annotation will affect any columns added later, unless their column-creation requests override the table-wide preferences.
 
+### Tag: 2019 Chaise Config
+
```
Comment (RFSH) [inline docs/user-docs/annotation.md]: Add the tag in a separate line here (like the rest of annotations)

File: docs/user-docs/annotation.md [hunk 2]
```diff
@@ -748,6 +749,23 @@ If a column-level annotation sets a _preference_ of `null`, this suppresses any
...
+This key indicates that the annotated catalog has a specific [chaise configuration](https://github.com/informatics-isi-edu/chaise/blob/master/docs/user-docs/chaise-config.md) that should override the default chaise-config properties and any properties defined in `chasie-config.js`. 
+
+Supported JSON payload patterns:
+The json object follows the same rules as [chaise-config.js](https://github.com/informatics-isi-edu/chaise/blob/master/chaise-config-sample.js). All properties can be defined at the root of the annotation, i.e:
+  - `{`... `"<chaise-config-property>"`: _value_ ...`}`
+
+The `chaise-config` property `configRules`, behaves the same way on the annotation that it does with the server wide config (`chaise-config.js`). The `configRules` will be checked for a match and apply and use those `chaise-config` properties over any other values defined for that same property. The order that the properties will be applied are as follows:
+  1. default values defined in [chaise-config.md](https://github.com/informatics-isi-edu/chaise/blob/master/docs/user-docs/chaise-config.md)
```
Comment (RFSH) [inline docs/user-docs/annotation.md]: I'm not sure what's the best way to explain it. I feel like Saying "The order... are as follows" suggests that the first rule will be applied and not the rest. Which is the opposite of how it's actually behaving.

**Comment by jrchudy** [comment]
@RFSH  this is also ready for review again after updating the documentation. I tried to make it a little more clear, but it still reads somewhat the same. If you think it's still confusing, I can ask Hongsuda for her suggestion.

**Comment by RFSH** [comment]
@jrchudy Thanks for adding more information. It's more clear now. But I think it's a good idea to ask @hongsudt to take a look at it too.

**Pull request merged**